### ISSUE TITLE
Additional extension methods

### ISSUE THREAD

**Issue by Jericho** [Pathoschild/FluentHttpClient#47]
I wrote a few simple extension methods that I'm using in my project. For the most part, they are simple one-liners that make it easier to perform repetitive tasks.

Let me know if you think it would be useful to merge them into FluentHtpClient.

```CSharp
// The APIs I interact with accept JSON 
public static IRequest WithJsonBody<T>(this IRequest request, T body);

// Use the PATCH method when dispatching a request
public static IRequest PatchAsync(this Pathoschild.Http.Client.IClient client, string resource);

// This solves the issue I discussed in #20. At the time you were not interested in implementing a fix, but I'm mentioning it again in case you changed your mind ;-)
public static Task<string> AsString(this IResponse response, Encoding encoding);
public static async Task<string> AsString(this IRequest request, Encoding encoding);
```

**Comment by Pathoschild** [comment]
method | comments
:----- | :-------
`WithJsonBody<T>` | ✘ Equivalent to`WithBody<T>`, since JSON is the default format.
`PatchAsync` | ✓ Feel free to submit a PR.
`AsString` | ✘ I still consider this a niche edge case that's better handled with a response filter. :)

**Comment by Jericho** [comment]
**PatchAsync**: done. PR submitted

**WithJsonBody**: Are you saying that I can invoke WithBody, omit the media type and `application/json` will be assumed? That's interesting, I didn't know.

**AsString**: I figured it was worth mentioning it again ;-)

**Comment by Pathoschild** [comment]
**PatchAsync:** merged into `develop` via #48 for the upcoming 3.0 release. Thanks!

**WithJsonBody:** yep, if you don't specify a body media type it'll default to the main formatter (i.e. the first one in the list).
